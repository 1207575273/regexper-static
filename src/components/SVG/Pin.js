import React from 'react';
import PropTypes from 'prop-types';

import style from './style';

import reflowable from './reflowable';

@reflowable
class Pin extends React.PureComponent {
  static defaultProps = {
    radius: 5
  }

  reflow() {
    const { radius } = this.props;

    this.setBBox({
      width: radius * 2,
      height: radius * 2
    });
  }

  render() {
    const { radius } = this.props;

    const circleProps = {
      r: radius,
      style: style.pin,
      transform: `translate(${ radius } ${ radius })`
    };

    return <circle { ...circleProps }></circle>;
  }
}

Pin.propTypes = {
  radius: PropTypes.number
};

export default Pin;
