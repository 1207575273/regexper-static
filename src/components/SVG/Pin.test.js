import React from 'react';
import { mount } from 'enzyme';

import Pin from './Pin';

describe('Pin', () => {
  test('rendering', async () => {
    const component = mount(
      <Pin/>
    );
    await component.instance().doReflow();
    component.update();
    expect(component).toMatchSnapshot();
  });
});
