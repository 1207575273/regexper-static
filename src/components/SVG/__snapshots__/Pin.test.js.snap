// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pin rendering 1`] = `
<Pin
  radius={5}
>
  <circle
    r={5}
    style={
      Object {
        "fill": "#6b6659",
        "stroke": "#000",
        "strokeWidth": "2px",
      }
    }
    transform="translate(5 5)"
  />
</Pin>
`;
